plugins/www/Optimove.js

```diff
@@ -6,3 +6,29 @@ var exec = require('cordova/exec');
 window.initialize = function (inAppConsentStrategy, success, error) {
     exec(success, error, 'OptimoveSDKPlugin', 'initialize', [inAppConsentStrategy]);
 };
+window.setUserId = function (userId, success, error) {
+    exec(success, error, 'OptimoveSDKPlugin', 'setUserId', [userId]);
+}
+window.setUserEmail = function (userEmail, success, error) {
+    exec(success, error, 'OptimoveSDKPlugin', 'setUserEmail', [userEmail]);
+};
+
+/* eventParams is nullable*/
+window.reportEvent = function (eventName, eventParams, success, error) {
+    exec(success, error, 'OptimoveSDKPlugin', 'reportEvent', [eventName, eventParams]);
+}
+
+/* screenCategory parameter is nullable*/
+window.reportScreenVisit = function (screenName, screenCategory, success, error) {
+    exec(success, error, 'OptimoveSDKPlugin', 'reportScreenVisit', [screenName, screenCategory]);
+}
+window.registerUser = function (userId, userEmail, success, error) {
```

Comment:
Can we use a Promise style API rather than taking success/error callbacks as method parameters?